### PR TITLE
Update to Twig 3

### PR DESCRIPTION
**Pull request by WyriHaximus**

### CODE REVIEW

**Comment by othercorey** [comment]
Can you remove and ignore the composer lock?

**Comment by WyriHaximus** [comment]
No, composer ignores it when you require a package.

File: composer.json [hunk 1]
```diff
@@ -19,11 +19,10 @@
...
-        "asm89/twig-cache-extension": "^1.0",
         "cakephp/cakephp": "4.x-dev as 4.0.0",
-        "jasny/twig-extensions": "^1.2.0",
-        "twig/twig": "^2.4.3"
+        "jasny/twig-extensions": "^1.3",
+        "twig/markdown-extra": "^3.0",
+        "twig/twig": "^3.0"
     },
     "require-dev": {
         "cakephp/cakephp-codesniffer": "dev-next",
```
Comment (ADmad) [inline composer.json]: Please change this also to `^2.0`

**Comment by othercorey** [comment]
> No, composer ignores it when you require a package.

Why no?  This doesn't test the same thing that requiring the package does.

**Comment by WyriHaximus** [comment]
> > No, composer ignores it when you require a package.
> 
> Why no? This doesn't test the same thing that requiring the package does.

This tests lowest, current, and highest possible for dependencies so it needs the lock.

File: composer.json (hunk 1 above)
Comment (WyriHaximus) [reply composer.json]: Done :+1:

**Comment by othercorey** [comment]
> This tests lowest, current, and highest possible for dependencies so it needs the lock.

Can you explain what 'current' catches that highest and lowest doesn't?

**Comment by WyriHaximus** [comment]
> > This tests lowest, current, and highest possible for dependencies so it needs the lock.
> 
> Can you explain what 'current' catches that highest and lowest doesn't?

Current is your guaranteed to work with set of versions. (Hence committing your lock file.) Where lowest and highest go for both outer boundaries of your dependencies looking for potential issues with your dependencies. Depending on the outcome of those you might need to change version targetting.

**Comment by WyriHaximus** [comment]
Tagging tomorrow morning

**Pull request merged**